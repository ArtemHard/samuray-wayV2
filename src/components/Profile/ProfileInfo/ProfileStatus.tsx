import React, { Component } from "react";

type ProfileStatusPropsType = {
  status: string;
  updateStatus: (newStatus: string) => void;
};

export class ProfileStatus extends Component<ProfileStatusPropsType> {
  state = {
    editMode: false,
    status: this.props.status,
  };

  activateEditMode = () => {
    this.setState({
      editMode: true,
    });
  };
  dectivateEditMode = () => {
    this.setState({
      editMode: false,
    });
    this.props.updateStatus(this.state.status);
  };
  onStatusChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      status: e.currentTarget.value,
    });
  };
  componentDidUpdate(
    prevProps: Readonly<ProfileStatusPropsType>,
    prevState: Readonly<{}>,
    snapshot?: any
  ): void {
    if (prevProps.status !== this.props.status) {
      this.setState({
        status: this.props.status,
      });
    }
  }
  render() {
    return (
      <div>
        {!this.state.editMode && (
          <div>
            <b>Status: </b>
            <span onDoubleClick={this.activateEditMode}>
              {this.props.status || "status doesnt exist"}
            </span>
          </div>
        )}
        {this.state.editMode && (
          <div>
            <input
              // ref={this.statusInputRef}
              onChange={this.onStatusChange}
              autoFocus={true}
              onBlur={this.dectivateEditMode}
              value={this.state.status}
            />
          </div>
        )}
      </div>
    );
  }
}
