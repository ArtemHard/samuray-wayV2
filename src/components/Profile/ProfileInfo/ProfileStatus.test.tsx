import TestRenderer, { ReactTestRenderer, create } from "react-test-renderer";
import { ProfileStatus } from "./ProfileStatus";

describe("ProfileStatus Component", () => {
  test("status from props should be in the state", () => {
    const testRenderer = TestRenderer.create(
      <ProfileStatus status='it-icuba-artem' updateStatus={() => {}} />
    );
    const tree = create(
      <ProfileStatus status='it-icuba-artem' updateStatus={() => {}} />
    );
    const testInstance = testRenderer.root;
    expect(testInstance.instance.state?.status).toEqual("it-icuba-artem");
    expect(tree).toMatchSnapshot();
  });
  describe("render title", () => {
    it("renders with  title", () => {
      const tree = create(
        <ProfileStatus status='it-icuba-artem' updateStatus={() => {}} />
      );
      expect(tree).toMatchSnapshot();
    });
    it("renders with new props", () => {
      const tree = create(
        <ProfileStatus status='it-icuba-123' updateStatus={() => {}} />
      );
      expect(tree).toMatchSnapshot();
    });
  });
  test("agter creation <span> should be displayed with correct status ", () => {
    const testRenderer = TestRenderer.create(
      <ProfileStatus status='it-icuba-artem' updateStatus={() => {}} />
    );

    const testInstance = testRenderer.root;
    // шляпа
    // const span = testInstance.findByType();
    // console.log("asdfasdf>>>>>>>>>>>>>>>.", span);

    // expect(testInstance.instance.state?.status).toEqual("it-icuba-artem");
  });
});
