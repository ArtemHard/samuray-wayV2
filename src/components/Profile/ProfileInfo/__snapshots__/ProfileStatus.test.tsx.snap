// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileStatus Component render title renders with  title 1`] = `
<div>
  <div>
    <span
      onDoubleClick={[Function]}
    >
      it-icuba-artem
    </span>
  </div>
</div>
`;

exports[`ProfileStatus Component render title renders with new props 1`] = `
<div>
  <div>
    <span
      onDoubleClick={[Function]}
    >
      it-icuba-123
    </span>
  </div>
</div>
`;

exports[`ProfileStatus Component status from props should be in the state 1`] = `
<div>
  <div>
    <span
      onDoubleClick={[Function]}
    >
      it-icuba-artem
    </span>
  </div>
</div>
`;
